signer/tests/integration/postgres.rs

```diff
@@ -1565,39 +1562,93 @@ async fn get_swept_deposit_requests_does_not_return_deposit_requests_with_respon
     // can rely on that fact.
+    setup_other.store_deposit_tx(&db).await;
     setup.store_deposit_tx(&db).await;
 
     // We take the sweep transaction as is from the test setup and
     // store it in the database.
+    setup_other.store_sweep_tx(&db).await;
     setup.store_sweep_tx(&db).await;
 
     // The request needs to be added to the database. This stores
     // `setup.deposit_request` into the database.
+    setup_other.store_deposit_request(&db).await;
     setup.store_deposit_request(&db).await;
 
-    // Here we store an event that signals that the deposit request has been confirmed.
+    // TODO: Create the initial transaction sweep package without any
+    // withdrawals and have a separate method for creating that sweep (since
+    // it's not realistic to have the withdrawal in the same sweep as the
+    // deposit). Then we wouldn't have to do this.
+    setup_other.store_withdrawal_request(&db).await;
+    setup.store_withdrawal_request(&db).await;
+
+    // Store outstanding sweep transaction packages in the database.
+    setup_other.store_sweep_transactions(&db).await;
+    setup.store_sweep_transactions(&db).await;
+
+    // Setup the stacks blocks
+    let stacks_tip = db.get_stacks_chain_tip(&chain_tip).await.unwrap().unwrap();
+
+    let setup_other_event_block = StacksBlock {
+        block_hash: fake::Faker.fake_with_rng(&mut rng),
+        block_height: stacks_tip.block_height + 1,
+        parent_hash: stacks_tip.block_hash,
+        // For `setup_other`, the stacks block is not in the canonical chain
+        bitcoin_anchor: fake::Faker.fake_with_rng(&mut rng),
+    };
+    let setup_event_block = StacksBlock {
+        block_hash: fake::Faker.fake_with_rng(&mut rng),
+        block_height: stacks_tip.block_height + 1,
+        parent_hash: stacks_tip.block_hash,
+        // For `setup`, the stacks block is in the canonical chain
+        bitcoin_anchor: chain_tip,
+    };
+    db.write_stacks_block_headers(vec![
+        setup_other_event_block.clone(),
+        setup_event_block.clone(),
+    ])
+    .await
+    .unwrap();
+
+    // First, let's check we get both deposits
+    let requests = db
+        .get_swept_deposit_requests(&chain_tip, context_window)
+        .await
+        .unwrap();
+
+    assert_eq!(requests.len(), 2);
+
+    // Here we store some events that signals that the deposit request has been confirmed.
     let event = CompletedDepositEvent {
         txid: fake::Faker.fake_with_rng::<StacksTxId, _>(&mut rng).into(),
-        block_id: fake::Faker
-            .fake_with_rng::<StacksBlockHash, _>(&mut rng)
-            .into(),
+        block_id: *setup_event_block.block_hash,
         amount: setup.deposit_request.amount,
         outpoint: setup.deposit_request.outpoint,
     };
-
     db.write_completed_deposit_event(&event).await.unwrap();
 
-    let chain_tip = setup.sweep_block_hash.into();
-    let context_window = 20;
+    let event = CompletedDepositEvent {
```

Comment:
suggestion: maybe add a comment to these two events to make it clear which is on the canonical chain and which isn't and the expected effect.

Reply:
Added comments and renamed things to make it clear which is which in [c3e0400](https://github.com/stacks-network/sbtc/pull/747/commits/c3e04003d7fb9578c919756d7ed254180c3df45a)